Exercise_4.py

```diff
@@ -1,12 +1,47 @@
+# Time Complexity : Merging two array of size n has a time complexity of O(n) and we divide the array log n time. Hence over time complexity is O(nlogn)
+# Space Complexity : 
```

Comment:
Space is O(N) as you are creating partitions of array and storing them in an array. Also the recursive stack will be using space under the memory.